rust/src/domains/polars/expr/mod.rs

```diff
@@ -69,6 +74,13 @@ impl ExprContext {
         }
     }
 
+    /// # Proof Definition
+    /// Only return without an error if the expression context `self`
+    /// allows for row ordering or the number of rows to be changed.
+    ///
+    /// This is used to disallow operations like sorting, shuffling or filtering
+    /// when row alignment must not change,
+    /// like when augmenting a new column via `with_column`.
     pub fn break_alignment(&self) -> Fallible<()> {
```

Comment:
Not the subject of this PR, but would this be more clear with a `check_` name?

Reply:
Hmm, it could be something like, `check_alignment_can_be_broken`.